Frontend/src/components/pages/Company.jsx

```diff
@@ -150,6 +154,52 @@ export default function Company() {
     setLoading(false);
   }
 
+  async function handleEditCompany() {
+      setCompany(company);
+      setOpenCompanyFormModal(true);
+  }
+
+  function navigateCompanies() {
+      navigate("/companies")
+  }
+
+  async function handleDeleteCompany() {
+      setObject({ type: "Company", name: company.name });
+      setEndpoint("/companies/" + company.id);
+      setPopulateObjects({ function: navigateCompanies });
+
+      setOpenDeleteAlert(true);
+  }
+
+  async function handleSoftLockCompany() {
+    setLoading(true);
+    const JWToken = JSON.parse(localStorage.getItem("loginInfo")).JWT;
+    try {
+      const serverResponse = await fetch(
+        "/companies/softLock/" + company.id,
```

Comment:
Ovdje treba malo popraviti. Znaci ne postoji taj endpoint nego samo companies/{id} i onda na taj endpoint saljes cijeli company object koji ukljucuje softLocked atribut.
Ovo treba i na backendu onda ispraviti.

Reply:
Zašto? Ovako je poprilično jasno što tražim, a ne da ljudi trebaju ići na backend da shvate koji vrag ovdje radim. Također, put, get i post su svi bili zauzeti za companies/{id}. Previše podataka slati je uvijek loše, a ne vidim poantu slanja cijelog company. Mislio sam možda pozivati edit, ali ovako mi se činilo bolje, budući da je softlock zasebna akcija.

Reply:
Pročitao sam donju poruku i shvatio da si sverojatno mislio da koristim edit. Svi komentari gore stoje, samo sada shvaćam prijedlog. Kao što sam napomenuo iznad, još uvijek sam za zaseban softlock.

Reply:
Znam da je previse podataka lose slati, ali svi zahtjevi u ovom aplikaciji se mjere kilobajtima tako da se ne moze osjetiti razlika.
A ako saljemo sve na vec postojeci PUT /api/company/{id} endpoint aplikacija je puno jednostavnija pa zato zelim da tamo saljemo.
Ako idemo tako zasebne endpointe onda ih mozemo dodati jos 10+ za svaku sitnicu koja se mijenja koja ne ide kroz formu, od statusa suradnji i redoslijeda kontakata do project members itd. Ovako vec imas pristup cijelom objektu, samo mu azuriraj zeljeni atribut i posalji sve na backend da se sve spremi.

Reply:
Samo da ostavim komentar sta smo dogovorili: ipak ce ici zaseban endpoint za to i definiran je u [endpoints](https://github.com/orgs/best-zagreb/projects/7?pane=issue&itemId=32159485): 
`PATCH /api/projects/{id}/softLock`
Auth: administrator or moderator (creator of that project) or projectResp (for that project)